compiler-core/src/language_server/engine.rs

```diff
@@ -154,7 +154,12 @@ where
                 None => return Ok(None),
             };
 
-            let location = match node.definition_location() {
+            let location = match node.definition_location(node.annotation_type().and_then(|t| {
+                type_constructor_from_modules(
+                    this.compiler.project_compiler.get_importable_modules(),
+                    t,
+                )
+            })) {
```

Comment:
Could we instead pass in the importable modules? That would make this API simpler, it would remove some of the extra work done in the cases when we don't use the type constructor information to find the location, and it would avoid having to pattern match on the node twice (once in definition_location, once in annotation_type).